tests/test_raster.py

```diff
@@ -940,29 +942,63 @@ def test_getitem_setitem(self, example: str) -> None:
         # The rasters should be the same
         assert rst2.raster_equal(rst)
 
-        # Check that errors are raised for both indexing and index assignment
+        # -- Second, we test NumPy indexes (slices, integers, ellipses, new axes) --
+
+        # Indexing
+        assert np.ma.allequal(rst[0], rst.data[0])  # Test an integer
+        assert np.ma.allequal(rst[0:10], rst.data[0:10])  # Test a slice
+        assert np.ma.allequal(rst[np.newaxis, 0], rst.data[np.newaxis, 0])  # Test a new axis
+        assert np.ma.allequal(rst[...], rst.data[...])  # Test an ellipsis
+
+        # Index assignment
+        rst[0] = 1
+        assert np.ma.allequal(rst.data[0], np.ones(np.shape(rst.data[0])))  # Test an integer
+        rst[0:10] = 1
+        assert np.ma.allequal(rst.data[0:10], np.ones(np.shape(rst.data[0:10])))  # Test a slice
+        rst[0, np.newaxis] = 1
```

Comment:
I have troubles understanding what this operation will do. Since `rst` has a dimension of 2 or 3, the second axis already exists... I assume it is equivalent to `rst[0] = 1` ?

Reply:
It's adding a dimension, but because I only index 0 in the first one, it returns a 2D raster of shape (height, 1). Added a comment!